extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/deployAzureSQLDBWizard.ts

```diff
@@ -172,4 +189,33 @@ export class DeployAzureSQLDBWizard extends WizardBase<DeployAzureSQLDBWizard, W
 			level: azdata.window.MessageLevel.Error
 		};
 	}
+
+	public async refreshPages() {
+
+		const pageCount = this.wizardObject.pages.length;
+		for (let i = 1; i < pageCount; i++) {
+			this.wizardObject.removePage(this.wizardObject.pages.length - 1);
+			this.wizardObject.pages.pop();
+		}
+		if (instanceOfAzureSQLDBDeploymentProvider(this.resourceProvider)) {
+			this._wizardInfo = this.resourceProvider.azureSQLDBWizard!;
+		} else {
+			return;
+		}
+
+		const newPages = this.getPages();
+
+		newPages[0] = this.pages[0];
+
+		this.pages = newPages;
+
+		for (let i = 1; i < newPages.length; i++) {
```

Comment:
This should not be needed in every wizard, it should be implemented in WizardBase as it was and all child classes just execute this code up the chain with appropriate super call.